scrypto/src/core/network.rs

```diff
@@ -1,42 +1,62 @@
 use core::str::FromStr;
+use sbor::rust::string::String;
 use sbor::{Decode, Encode, TypeId};
 
-/// Generates an Enum and defines its `FromStr` implementation.
-macro_rules! network_enum_with_from_str {
-    (
-        $(#[$meta:meta])*
-        $vis:vis enum $enum_name:ident {
-            $($variant:ident),*
-        }
-    ) => {
-        $(#[$meta])*
-        $vis enum $enum_name{
-            $(
-                $variant,
-            )*
-        }
+/// Network Definition is intended to be the actual definition of a network
+#[derive(Debug, Clone, TypeId, Encode, Decode, PartialEq, Eq)]
+pub struct NetworkDefinition {
+    pub id: u8,
+    pub name: String,
+    pub hrp_suffix: String,
+}
 
-        impl FromStr for $enum_name {
-            type Err = NetworkError;
-
-            fn from_str(s: &str) -> Result<Self, Self::Err> {
-                match s {
-                    $(
-                        stringify!($variant) => Ok(Self::$variant),
-                    )*
-                    _ => Err(NetworkError::InvalidNetworkString),
-                }
-            }
+// TODO: we may be able to squeeze network identifier into the other fields, like the `v` byte in signature.
+/// Network is intended to be an "easy" method to get common Network configuration for (eg) tests
+#[derive(Debug, Clone, TypeId, Encode, Decode)]
+pub enum Network {
+    LocalSimulator,
+    Mainnet,
+    Custom(NetworkDefinition),
+}
```

Comment:
I'm not sure if this enum is necessary. And the abstraction doesn't seem right (there is no logical separation between the three variants: `Custom` can be `Mainnet`).

For convenience, I would consider using helper functions: `NetworkDefinition::mainnet()`  returns the definition of mainnet.

Reply:
Yeah, that's a good shout - I'll make this change :)

Reply:
Also would consider having network ID as single source of truth and deriving name and HRP from id.

```rust
struct Network(u8);

impl Network {
     pub fn id() -> u8;
     pub fn name() -> String;  // computed based on id
     pub fn hrp() -> String;  // computed based on id
}
```

Reply:
Well that would depend on where the source of truth lives for the Network Definition.

Chatting to Josh, we seemed to think the Node could/should own it for now? 👍

Reply:
Cool